### PR TITLE
Fix VS becomes unresponsive opening large project

### PR DESCRIPTION
**Pull request by davkean**
**Customer scenario**

After installing one of the 15.3 Previews customers will start experiencing more "random" UI delays in normal VS operations including:

- Expanding nodes in Solution Explorer
- Add new and removing items to and from the tree
- Opening C# source files
- Opening XML files
- Typing
- Opening and closing the solution

**Bugs this fixes:** 

Part of the general effort of chasing down the UI delays introduced in the project system in 15.3: https://github.com/dotnet/project-system/issues/2297.

This particular issue is being tracked by: https://github.com/dotnet/project-system/issues/2383.

**Workarounds, if any**

None

**Risk**

Low.

**Performance impact**

This will reduce UI delays and reduce the number of active threads.

**Is this a regression from a previous update?**

Yes, the dependency node was rewritten in 15.3.

**Root cause analysis:**

We were hitting an [issue in JoinableTaskFactory](https://github.com/Microsoft/vs-threading/issues/132) which resulted in a bunch of threads blocked on a single thread. This pattern is unusual and is something we only hit - we felt that a change to us was less risky for 15.3 to VS than to the JoinableTaskFactory which is used by all of VS.

**How was the bug found?**

Many internal reports both from ASP.NET, Roslyn, ProjectSystem itself and opening external projects (such as https://github.com/aarnott/pinvoke). Also had external reports from MVPs.

### CODE REVIEW

**Comment by davkean** [comment]
This is a work in progress - I've not fully finished testing this.

**Comment by davkean** [comment]
I've verified this change along with #2429, #2410, #2435 and #2400.

tag @MattGertz for 15.3 consideration.

**Comment by davkean** [comment]
tag @dotnet/project-system @lifengl for review

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs [hunk 1]
```diff
@@ -128,34 +128,14 @@ Task ILanguageServiceHost.InitializeAsync(CancellationToken cancellationToken)
             }
         }
 
-        private JoinableTask<T> ExecuteWithinLockAsync<T>(Func<Task<T>> task)
+        private Task<T> ExecuteWithinLockAsync<T>(Func<Task<T>> task)
```
Comment (natidea) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs]: >Task<T> [](start = 16, length = 7)

With this change, we now need a ConfigureAwait on lines 208, 270 and 291

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs [hunk 2]
```diff
@@ -321,34 +321,14 @@ private void DisposeAndClearSubscriptions()
             }
         }
 
-        private JoinableTask<T> ExecuteWithinLockAsync<T>(Func<Task<T>> task)
+        private Task<T> ExecuteWithinLockAsync<T>(Func<Task<T>> task)
```
Comment (natidea) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs]: >Task<T> [](start = 16, length = 7)

With this change, we now need a ConfigureAwait on lines 68, 76, 158, 230 and 302

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs (hunk 1 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs]: Fixed these, appreciate you telling the exact lines - I'm not getting the build errors in the parent commit this applied to.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs (hunk 2 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs]: Fixed.

**Pull request merged**